source/MaterialXGenShader/ShaderStage.h

```diff
@@ -114,13 +114,14 @@ class MX_GENSHADER_API VariableBlock
 
     /// Add a new shader port to this block.
     /// @param type The desired shader port type
+    /// @param context The GenContext object being used
     /// @param name The shader port name
     /// @param value The value to attach to the shader port
     /// @param shouldWiden When false, an exception is thrown if the type of the existing port with
     ///   the same name does not match the requested type. When true, the types can mismatch, and the
     ///   type of any existing port is widened to match the requested type when necessary.
     /// @return A new shader port, or a pre-existing shader port with the same name.
-    ShaderPort* add(TypeDesc type, const string& name, ValuePtr value = nullptr, bool shouldWiden = false);
+    ShaderPort* add(TypeDesc type, const GenContext& context, const string& name, ValuePtr value = nullptr, bool shouldWiden = false);
```

Comment:
I'm happy to conform that, I was wary of making too many API changes all at once.  Perhaps we get this update to land - and then I can take a look at a separate PR that makes the API feel more consistent.  I think there's also scope to look at passing GenContext as a `const GenContext&` in more places too perhaps.

Reply:
Interested to hear from @jstone-lucasfilm what his preference is here.  Defer API changes - or roll them all in to the same PR?

Reply:
I was mainly referring to just the API functions that were changed here anyway. Not to change this in all API. That could definitely wait for a later PR.

But this was just an idea for consistency. Perhaps it becomes less consistent anyway because of how it is passed in other places in the API. It’s just coding style and not a big issue.

Reply:
I do like the idea of making it consistent as a secondary PR.